dashboard/lib/queries/user_progress.rb

```diff
@@ -0,0 +1,7 @@
+class Queries::UserProgress
+  def self.user_levels_by_level(user, script)
```

Comment:
I'd like to take this opportunity to give this a more accurate name, but I'm not sure what; I'd like to be able to capture both that these are the user levels for a given script and that they are indexed by the level's _id_ (rather than the level itself). Something like `user_levels_for_script_by_level_id`, but less unwieldy, 

I almost wonder if it would be easier to just do `UserLevels.where(user: user, script: script).index_by(&:level_id)`

Reply:
Related: https://github.com/code-dot-org/code-dot-org/pull/30990 and https://codedotorg.slack.com/archives/C0T0PNTM3/p1569538451177600

Reply:
Either `user_levels_for_script_by_level_id` or `UserLevels.where(user: user, script: script).index_by(&:level_id)` seem fine to me. Either one leaves the code better than we found it and will be easy to change later if a different naming policy is adopted . @Hamms , especially since you've been working on cleaning up this code, I'd be very receptive to any guidelines that might make it easier to make incremental naming cleanups in the future.

Reply:
Where would be the right place to capture those guidelines, do you think?

Reply:
No strong preference. If you do this it would be fine for any guidelines to be informally documented in code review discussions as well as the resulting code. I would be receptive to a vision for naming in this file but otherwise I'd assume we're just making a best effort to make code more readable as we go.